src/nomad_simulation_parsers/parsers/h5md/parser.py

```diff
@@ -0,0 +1,360 @@
+            return []
+
+        source_data = self.get_source(self.data, kwargs['path'])
+        include = kwargs.get('include')
+        exclude = kwargs.get('exclude')
+        contributions = []
+        for key, val in source_data.items():
+            if include and key not in include or exclude and key in exclude:
+                continue
+            step_data = self.get_step_data(val, source['step'])
+            contributions.append({'name': key, **step_data})
+        return contributions
+
+    def get_output_data(self, source: dict[str, Any], **kwargs) -> pint.Quantity | None:
+        if source.get('value') is not None:
+            return source['value']
+        if source.get('step') is None:
+            return
+        observable_type = kwargs.get('observable_type')
+        if observable_type is None or observable_type not in [
+            'configurational',
+            'ensemble_average',
+            'correlation_function',
+        ]:
+            self.logger.warning(
+                'Invalid or no obervable type defined in the schema annotation, '
+                'skipping this observable.'
+            )
+            return
+
+        source_data = self.get_source(self.data, kwargs['path'])
+        if source_data.get('@type') != observable_type:
+            return
+
+        data = self.get_step_data(source_data, source['step']).get('value')
+
+        return data
+
+    def get_custom_outputs(
+        self, source: dict[str, Any], **kwargs
+    ) -> list[dict[str, Any]]:
+        if source.get('step') is None:
+            return []
+
+        source_data = self.get_source(self.data, kwargs['path'])
+        include = kwargs.get('include')
+        exclude = kwargs.get('exclude')
+        observable_type = kwargs.get('observable_type')
+        custom_outputs = []
+        for key, val in source_data.items():
+            if include and key not in include or exclude and key in exclude:
+                continue
+            if observable_type is not None:
+                source_type = val.get('@type')
+                if source_type != observable_type:
+                    continue
+            step_data = self.get_step_data(val, source['step'])
+            if step_data.get('value') is not None:
+                if isinstance(step_data['value'], pint.Quantity):
+                    step_data['unit'] = str(step_data['value'].units)
+                    step_data['value'] = step_data['value'].magnitude
+            custom_outputs.append({'name': key, **step_data})
+        return custom_outputs
+
+
+class H5MDArchiveWriter(ArchiveWriter):
+    def __init__(self):
+        self.h5_parser = H5MDH5Parser()
+        self.simulation_parser = H5MDMetainfoParser()
+        self.simulation_parser.max_nested_level = 10
+        self.workflow_parser = H5MDMetainfoParser()
+
+    def write_to_archive(self) -> None:
+        # reload schema annotations
+        reload(h5md)  # ? what exactly does this do?
```

Comment:
this is necessary as all the annotations are written in the general schema, so when another parser is executed the h5md annotations would have been overwritten. top-level imports are cached so a brute force reload is necessary.